### PR TITLE
BugFix: Add check for color scheme to ensure correct favicon is updated

### PR DESCRIPTION
**Pull request by zhen0**
Our new favicons are color scheme/appearance (light/dark mode) aware so we now have more than one 'favicon-32' and 'favicon-16'. This PR adds a check for color mode and selects and updates the correct favicon. 

Partner to https://github.com/PrefectHQ/prefect/pull/9370 and https://github.com/PrefectHQ/nebula-ui/pull/2706 which set the correct id based on color scheme selection.

### CODE REVIEW

File: src/compositions/useFavicon.ts [hunk 1]
```diff
@@ -1,9 +1,33 @@
...
+      favicon32 = document.getElementById('favicon-32-dark')
+      break
+    case 'light':
+      favicon16 = document.getElementById('favicon-16-light')
+      favicon32 = document.getElementById('favicon-32-light')
+      break
+    default:
+      favicon16 = document.getElementById('favicon-16-no-preference')
+      favicon32 = document.getElementById('favicon-32-no-preference')
+      break
```
Comment (pleek91) [inline src/compositions/useFavicon.ts:29]: We have a specific favicon for no preference?

File: src/compositions/useFavicon.ts [hunk 2]
```diff
@@ -1,9 +1,33 @@
...
 import { StateType } from '@/models/StateType'
 
+function getPreferredColorScheme(): 'dark' | 'light' | 'no-preference' {
+  if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
+    return 'dark'
+  } if (window.matchMedia('(prefers-color-scheme: light)').matches) {
+    return 'light'
+  }
+  return 'no-preference'
+}
```
Comment (pleek91) [inline src/compositions/useFavicon.ts:11]: Should this be based on the prefect-design mode? That way if a user switches their color settings this reflects that change.

Comment (znicholasbrown) [reply src/compositions/useFavicon.ts:11]: Nope, this is a system setting - the background of the tab/window is what's important, not the application

File: src/compositions/useFavicon.ts (hunk 1 above)
Comment (znicholasbrown) [reply src/compositions/useFavicon.ts:29]: We do not - we default to the dark (official) one

File: src/compositions/useFavicon.ts [hunk 3]
```diff
@@ -1,9 +1,33 @@
...
+  let favicon32: HTMLElement | null
+
+  switch (getPreferredColorScheme()) {
+    case 'dark':
+      favicon16 = document.getElementById('favicon-16-dark')
+      favicon32 = document.getElementById('favicon-32-dark')
+      break
+    case 'light':
+      favicon16 = document.getElementById('favicon-16-light')
+      favicon32 = document.getElementById('favicon-32-light')
```
Comment (znicholasbrown) [inline src/compositions/useFavicon.ts:24]:
```suggestion
  default:
      favicon16 = document.getElementById('favicon-16')
      favicon32 = document.getElementById('favicon-32')
```

File: src/compositions/useFavicon.ts (hunk 1 above)
Comment (znicholasbrown) [inline src/compositions/useFavicon.ts:29]:
```suggestion
```
Can remove these in favor of the case above

Comment (zhen0) [reply src/compositions/useFavicon.ts:29]: Any objection to removing the `prefers-color-scheme:no-preference` icon in the prefect and nebula-ui repos to go alongside this?  It looks like it's deprecated anyways and not supported anywhere but safari (which seems to be doing its own thing with favicons and ignoring the useFavicon change anyways).

https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme

Safari in light and dark when all other browsers are updating the favicon:
image

image

Comment (znicholasbrown) [reply src/compositions/useFavicon.ts]: No objection from me - Safari might handle favicons a little differently, we've got a `mask-icon` link in the head which might be what Safari is pulling instead but I'm not sure

**Review by znicholasbrown — APPROVED** [review]

**Pull request merged**